projects/copyleaks-web-report/src/lib/components/containers/content-ai-container/components/explainable-ai-result-container/explainable-ai-result-container.component.ts

```diff
@@ -81,38 +84,47 @@ export class ExplainableAIResultContainerComponent implements OnInit, OnChanges
 	infoTooltipText: string;
 	proportionTooltipText: string;
 	resultTooltipText: string;
+	barTooltipText: string;
 	updateResult: boolean = false;
 	panelIndex: number[] = [];
 	tooltipVisible: boolean = false;
 
 	constructor() {}
 
 	ngOnChanges(changes: SimpleChanges): void {
-		if (changes.selectAIText) {
-			const selectResult = this.explainResults?.filter(item =>
-				this.selectAIText?.find(range => item.start <= range.start && range.end <= item.end)
-			);
-
-			if (!!this.selectAIText.length && !!selectResult.length) {
-				this.selectedMatch = true;
-				this.title =
-					`${selectResult.length} ` +
-					(selectResult.length > 1 ? $localize`AI Insights Selected` : $localize`AI Insight Selected`);
-				this.explainItemResults = [...selectResult];
-				this.closePanel();
-			} else {
-				this.selectedMatch = false;
-				this.title = $localize`AI Insights`;
-				this.explainItemResults = [...this.explainResults];
-			}
-		}
 		if (changes['isLoading']?.currentValue == false) {
 			this._initResults();
 		}
 	}
 
 	ngOnInit(): void {
 		if (!this.isLoading) this._initResults();
+		this.reportMatchesSvc.aiInsightsShowResult$.subscribe((selectResult: ISelectExplainableAIResult) => {
```

Comment:
@eyadh21 missing use of `untilDestroy `pipe for this subscription (`.pipe(untilDestroy(this))`)